### DISCUSSION TITLE
Use polling as a fallback

### DISCUSSION THREAD

**Discussion by quant-daddy** [enisdenjo/graphql-ws#217] [Q&A] [answered]
Sometimes websocket connection fails. Is there a way to use polling as a fallback on the browser? Also, is there a way detect if the websocket connection failed?

**Comment by enisdenjo** [comment, marked as answer]
This is a WebSocket library, so it implements just that. [Do note that WebSockets are widely supported.](https://caniuse.com/?search=websocket) Alternatively, you can try [`socket-io-graphql-client`](https://github.com/n1ru4l/graphql-live-query/tree/main/packages/socket-io-graphql-client) which uses [Socket.io](https://socket.io/) instead.

All errors and closures are reported by [`EventError`](https://github.com/enisdenjo/graphql-ws/blob/master/docs/modules/client.md#eventerror) and [`EventClosed`](https://github.com/enisdenjo/graphql-ws/blob/master/docs/modules/client.md#eventclosed) events respectively.
